services/brig/src/Brig/API.hs

```diff
@@ -111,7 +111,7 @@ sitemap o = do
     accept "application" "json"
       .&. jsonRequest @NewUser
   put "/i/self/email" (continue changeSelfEmailNoSendH) $
-    header "Z-User"
+    zauthUserId
```

Comment:
This slightly changes behaviour, as it masks the exact error message:

```
zheader = fmap (result (Fail . const accessDenied) Okay) . header
```

We already do the same in Galley, so it's probably a desirable change.